Build fails on macOS due to https://github.com/fish-shell/fish-shell/commit/d0a67e372c492ffc7c50efa838a19f59077e1de0:

```
  Scanning dependencies of target fish_indent
  [ 95%] Building CXX object CMakeFiles/fish_indent.dir/src/fish_indent.cpp.o
  [ 96%] Building CXX object CMakeFiles/fish_indent.dir/src/print_help.cpp.o
  [ 96%] Linking CXX executable fish_indent
  /nix/store/i9jh1yxfinl0pa5y2c9h2z5i0sdhaf7q-bash-4.4-p23/bin/bash: codesign: command not found
  make[2]: *** [CMakeFiles/fish_indent.dir/build.make:121: fish_indent] Error 127
  make[2]: *** Deleting file 'fish_indent'
  make[1]: *** [CMakeFiles/Makefile2:662: CMakeFiles/fish_indent.dir/all] Error 2
  make: *** [Makefile:147: all] Error 2
```

Since we presumably don't want to break the sandbox just for codesigning (can you even Ad-Hoc codesign without a developer account?) we're going to have to disable it. I'll file an upstream issue about having an official way to turn it off, but until then I guess we can just patch `CMakeLists.txt`.

--- a/pkgs/shells/fish/default.nix
+++ b/pkgs/shells/fish/default.nix
@@ -110,6 +110,10 @@ let
       sha256 = "1f12c56v7n4s0f9mi9xinviwj6kpwlcjwaig1d4vsk5wlgp7ip07";
     };
 
+    # We don't have access to the codesign executable, so we patch this out.
+    # For more information, see: https://github.com/fish-shell/fish-shell/issues/6952
+    patches = lib.optional stdenv.isDarwin ./dont-codesign-on-mac.diff;
+
     nativeBuildInputs = [
       cmake
     ];

--- a/pkgs/shells/fish/dont-codesign-on-mac.diff
+++ b/pkgs/shells/fish/dont-codesign-on-mac.diff
@@ -1,0 +1,12 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 7d220a032..786b60e6e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -183,7 +183,6 @@ ENDFUNCTION(CODESIGN_ON_MAC target)
+ # Define a function to link dependencies.
+ FUNCTION(FISH_LINK_DEPS_AND_SIGN target)
+   TARGET_LINK_LIBRARIES(${target} fishlib)
+-  CODESIGN_ON_MAC(${target})
+ ENDFUNCTION(FISH_LINK_DEPS_AND_SIGN)
+ 
+ # Define libfish.a.

